src/contexts/filterJobProvider.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useState, useEffect, createContext } from 'react';
+import { JobItem } from 'models/apiModels';
+export interface FilterContext {
+  allJobs: JobItem[];
+  sortedJobs: JobItem[];
+  tags: string[];
+  search_string: string;
+  sort_date: boolean;
+}
+type FilterHookType = {
+  context: FilterContext;
+  setContext: (_: FilterContext) => void;
+};
+
+export const FilterContextHook = createContext<FilterHookType>({
+  context: {
+    allJobs: [],
+    sortedJobs: [],
+    tags: [],
+    search_string: '',
+    sort_date: false,
+  } as FilterContext,
+  setContext: (_: FilterContext) => {},
+});
+
+const JobFilterProvider: React.FC = ({ children }) => {
+  const [filterContext, setFilterContext] = useState<FilterContext>({
+    allJobs: [],
+    sortedJobs: [],
+    tags: [],
+    search_string: '',
+    sort_date: false,
+  });
+
+  useEffect(() => {
```

Comment:
If this does not scale with many jobs, we can try to index once at startup :bulb:

Reply:
Can be resolved when we get too many job applications :))